bgs/bgs.go

```diff
@@ -661,3 +663,24 @@ func (s *BGS) createExternalUser(ctx context.Context, did string) (*models.Actor
 
 	return subj, nil
 }
+
+func (bgs *BGS) TakedownRepo(ctx context.Context, did string) error {
+	u, err := bgs.lookupUserByDid(ctx, did)
+	if err != nil {
+		return err
+	}
+
+	if err := bgs.db.Model(User{}).Where("id = ?", u.ID).Update("takedown", true).Error; err != nil {
+		return err
+	}
+
+	if err := bgs.repoman.TakedownRepo(ctx, u.ID); err != nil {
+		return err
+	}
+
+	if err := bgs.events.TakedownRepo(ctx, u.ID); err != nil {
+		return err
+	}
```

Comment:
at first I thought all of this should be in a transaction. but thinking about it more, I guess we probably want to update the user table pretty quick with the takedown state, and then proceed with the other steps, which take out locks. I think doing it in that order would prevent any obvious race conditions between the takedown API handler and any event stream processing?

And, admin/client can just send another takedown API request if the first does not complete, and that should be fine.

Reply:
yeah, the actions are idempotent, so can just keep rerunning it until it sticks if theres any errors.